server/resource_server/utils/utils.go

```diff
@@ -24,14 +24,18 @@ const SecretSize = 32
 
 var WorkingDirectory string
 
-// Response is used for static shape json return
 type Response struct {
```

Comment:
why not just use single type here and omitempty the error and data field?

Reply:
I think yes we can do that, will try that.

Reply:
I will try that.